addons/mail_client_extension/controllers/mail_plugin.py

```diff
@@ -15,32 +16,172 @@
 
 class MailPluginController(http.Controller):
 
+    @http.route('/mail_client_extension/modules/get', type="json", auth="outlook", csrf=False, cors="*")
+    def modules_get(self, **kwargs):
+        """
+            deprecated as of saas-14.3, not needed for newer versions of the mail plugin but necessary
+            for supporting older versions
+        """
+        return {'modules': ['contacts', 'crm']}
+
+    @http.route('/mail_plugin/partner/enrich_and_create_company',
+                type="json", auth="outlook", cors="*")
+    def res_partner_enrich_and_create_company(self, partner_id):
+        """
+        Route used when the user clicks on the create and enrich partner button
+        it will try to find a company using IAP, if a company is found
+        the enriched company will then be created in the database
+        """
+        response = {}
+
+        partner = request.env['res.partner'].browse(partner_id)
+
+        normalized_email = partner.email_normalized
+        if not normalized_email:
+            response = {'error': 'contact has no email'}
+            return response
+
+        company_domain = normalized_email.split('@')[1]
+
+        company, enrichment_info = self._create_company_from_iap(company_domain)
+
+        response['enrichment_info'] = enrichment_info
+        response['company'] = self._prepare_company_values(company)
+        if company:
+            partner.write({'parent_id': company})
+
+        return response
+
+    @http.route(['/mail_client_extension/partner/get', '/mail_plugin/partner/get']
+        , type="json", auth="outlook", cors="*")
+    def res_partner_get(self, email=None, name=None, partner_id=None, **kwargs):
```

Comment:
it is only the route name which is deprecated and not the entire method itself. I've added more explanation in the docstring to better explain this.

The code has to be changed to avoid creating companies for existing database partners (as this may lead to some undesirable behavior like having their invoices related to the new parent), furthermore, the method also allows to fetch a partner using an id.

All these changes were made in such a manner that allows older versions of the plugin to still work correctly with this route, that's why we still have to keep the /mail_client_extension route.